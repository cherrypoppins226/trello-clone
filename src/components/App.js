import React from "react";
import Board from "./Board";

const App = ({ lists }) => <Board lists={lists} />;

export default App;
