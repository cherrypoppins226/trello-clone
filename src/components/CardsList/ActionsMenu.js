import React from "react";
import PropTypes from "prop-types";
import { withStyles } from "material-ui/styles";
import Typography from "material-ui/Typography";
import Divider from "material-ui/Divider";
import { MenuList, MenuItem } from "material-ui/Menu";
import grey from "material-ui/colors/grey";

const styles = {
  heading: {
    paddingTop: 8,
    color: grey[700]
  },
  menu: {
    width: 300,
    "& *": {
      fontWeight: 700
    },
    "& hr": {
      margin: 5,
      marginLeft: 10,
      marginRight: 10
    },
    "& [role='menuitem']": {
      padding: 3,
      paddingLeft: 10,
      transition: "",
      "&:hover": {
        backgroundColor: "rgb(41, 143, 202)",
        "& p": { color: "white" }
      }
    }
  }
};

const ActionsMenu = ({ classes, onMenuItemClick }) => {
  return (
    <>
      <Typography role="heading" align="center" className={classes.heading}>
        List Actions
      </Typography>
      <MenuList className={classes.menu} dense={true} onClick={onMenuItemClick}>
        {[
          "",
          "Add Card...",
          "Copy List...",
          "Move List...",
          "Watch",
          "",
          "Sort By",
          "",
          "Move All Cards in This List...",
          "Archive All Cards in This List...",
          "",
          "Archive This List"
        ].map((item, idx) => {
          if (!Boolean(item)) return <Divider key={idx} />;
          return (
            <MenuItem key={idx}>
              <Typography> {item} </Typography>
            </MenuItem>
          );
        })}
      </MenuList>
    </>
  );
};

const View = withStyles(styles)(ActionsMenu);

View.propTypes = {
  onMenuItemClick: PropTypes.func.isRequired
};

export default View;
