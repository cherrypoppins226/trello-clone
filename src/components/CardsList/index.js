import React from "react";
import { withStyles } from "material-ui/styles";
import Button from "material-ui/Button";
import Paper from "material-ui/Paper";
import grey from "material-ui/colors/grey";
import PropTypes from "prop-types";
import MoreHoriz from "@material-ui/icons/MoreHoriz";
import Title from "./Title";
import Card from "./Card";
import * as Labels from "../labels";

const styles = {
  root: {
    display: "flex",
    flexDirection: "column",
    maxHeight: "100%",
    background: "rgb(226, 228, 230)",
    "& > button": {
      justifyContent: "left",
      textTransform: "none"
    },
    "& ul": {
      padding: 0,
      margin: 0,
      height: "100%",
      "& li": {
        margin: 8,
        marginTop: 2,
        cursor: "pointer"
      }
    }
  },
  listHeader: {
    display: "flex",
    flexShrink: 0,
    justifyContent: "space-between",
    margin: 5,
    "& [role='heading']": {
      flexBasis: "100%",
      fontWeight: 700,
      paddingBottom: 0,
      paddingTop: 3,
      paddingLeft: 5,
      marginRight: 8
    },
    "& p[role='heading']": {
      cursor: "pointer"
    },
    "& button": {
      alignSelf: "flex-start",
      // Reusable styles
      borderWidth: 0,
      background: "none",
      outline: "none",
      cursor: "pointer",
      borderRadius: 4,
      padding: 3,
      "&:hover": {
        background: "#D6D6D6"
      }
    },
    "& svg": {
      // Reusable styles
      padding: 3,
      width: "0.8em",
      height: "0.8em",
      color: grey[700]
    }
  }
};

class CardsList extends React.Component {
  constructor(props) {
    super(props);
    this.state = {
      counter: props.cards.length,
      cards: props.cards.map((card, idx) => ({
        id: idx,
        description: card
      }))
    };
  }

  addCard(e) {
    e.preventDefault();
    this.setState((prevState, props) => ({
      counter: prevState.counter + 1,
      cards: [
        ...prevState.cards,
        { id: prevState.counter + 1, description: undefined }
      ]
    }));
  }

  componentDidUpdate(prevProps, prevState) {
    if (prevState.cards.length < this.state.cards.length)
      this.cardsListEnd.scrollIntoView();
  }

  render() {
    const { classes } = this.props;
    return (
      <Paper
        component="section"
        elevation={1}
        className={classes.root}
        data-testid="CardsList"
      >
        <div className={classes.listHeader}>
          <Title role="heading" text={this.props.title} />
          <button
            onClick={e => this.props.onEditList(e.currentTarget)}
            aria-labelledby={Labels.cardsListActionsMenu.id}
            aria-haspopup={true}
          >
            <MoreHoriz />
          </button>
        </div>
        <div style={{ overflowY: "scroll" }}>
          <ul>
            {this.state.cards.map(({ id, description }) => (
              <li key={id}>
                <Card
                  description={description}
                  onQuickEditCard={this.props.onQuickEditCard}
                  onEditCard={this.props.onEditCard}
                />
              </li>
            ))}
          </ul>
          {/* The sole purpose of this div is to allow scrolling to bottom */}
          <div
            style={{ float: "left", clear: "both" }}
            ref={node => (this.cardsListEnd = node)}
          />
        </div>
        <Button onClick={this.addCard.bind(this)}>Add a card...</Button>
      </Paper>
    );
  }
}

const View = withStyles(styles)(CardsList);

View.propTypes = {
  title: PropTypes.string.isRequired,
  cards: PropTypes.array.isRequired,
  onEditList: PropTypes.func.isRequired,
  onQuickEditCard: PropTypes.func.isRequired,
  onEditCard: PropTypes.func.isRequired
};

export default View;
