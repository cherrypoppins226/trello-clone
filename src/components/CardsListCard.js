import React from "react";
import path from "path";
import { withStyles } from "material-ui/styles";
import Paper from "material-ui/Paper";
import Typography from "material-ui/Typography";
import PropTypes from "prop-types";
import grey from "material-ui/colors/grey";
import Create from "@material-ui/icons/Create";
import * as Labels from "./labels";

const styles = {
  container: {
    margin: 8,
    marginTop: 2,
    cursor: "pointer"
  },
  card: {
    padding: 4,
    paddingLeft: 8,
    display: "flex",
    flexDirection: "row",
    justifyContent: "space-between",
    "&:hover": {
      background: grey[200]
    },
    "&:hover button": {
      visibility: "visible"
    },
    "& button": {
      visibility: "hidden",
      // Reusable styles
      borderWidth: 0,
      background: "none",
      outline: "none",
      cursor: "pointer",
      alignSelf: "flex-start",
      borderRadius: 4,
      padding: 3,
      paddingBottom: 0,
      "&:hover": {
        background: "#D6DADC"
      }
    },
    "& svg": {
      // Reusable styles
      padding: 3,
      width: "0.8em",
      height: "0.8em",
      color: grey[700]
    }
  }
};

const onClick = (event, onEditCard, onEditFullCard) => {
  const button = event.currentTarget.querySelector(
    `[aria-labelledby="${Labels.editCard.id}"]`
  );
  if (button.contains(event.target)) {
    onEditCard(event.currentTarget);
  } else {
    onEditFullCard(event.currentTarget);
  }
};

export const cardDescription = cardNode => {
  return cardNode.querySelector(`[aria-labelledby="${Labels.card.id}"]`)
    .textContent;
};

const View = ({
  classes,
  description = "Title...",
  onEditCard,
  onEditFullCard
}) => {
  return (
    <li data-testid="CardsListCard" className={classes.container}>
      <Paper
        elevation={1}
        onClick={e => onClick(e, onEditCard, onEditFullCard)}
        className={classes.card}
        aria-labelledby={Labels.fullyEditCard.id}
      >
        <Typography aria-labelledby={Labels.card.id}>{description}</Typography>
        <button aria-labelledby={Labels.editCard.id}>
          <Create />
        </button>
      </Paper>
    </li>
  );
};

View.propTypes = {
  children: PropTypes.node,
  classes: PropTypes.object.isRequired,
  description: PropTypes.string,
  onEditCard: PropTypes.func.isRequired,
  onEditFullCard: PropTypes.func.isRequired
};

View.displayName = path.basename(__filename, path.extname(__filename));

export default withStyles(styles)(View);
