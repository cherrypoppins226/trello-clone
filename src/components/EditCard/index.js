import React from "react";
import PropTypes from "prop-types";
import { withStyles } from "material-ui/styles";
import * as Labels from "../labels";

const styles = {};

const EditCard = ({ classes, card = null, ...extra }) => {
  return <div aria-describedby={Labels.editCardDescription.id} {...extra} />;
};

const View = withStyles(styles)(EditCard);

View.propTypes = {
  card: PropTypes.object
};

export default View;
