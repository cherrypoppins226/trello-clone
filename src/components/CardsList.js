import React from "react";
import { findDOMNode } from "react-dom";
import path from "path";
import { withStyles } from "material-ui/styles";
import Button from "material-ui/Button";
import Typography from "material-ui/Typography";
import Paper from "material-ui/Paper";
import grey from "material-ui/colors/grey";
import PropTypes from "prop-types";
import MoreHoriz from "@material-ui/icons/MoreHoriz";
import TextArea from "./TextArea";
import CardsListCard from "./CardsListCard";
import * as Labels from "./labels";

const styles = {
  container: {
    display: "flex",
    flexDirection: "column",
    maxHeight: "100%",
    background: "rgb(226, 228, 230)",
    "& > button": {
      justifyContent: "left",
      textTransform: "none"
    },
    "& ul": {
      padding: 0,
      margin: 0,
      height: "100%"
    }
  },
  listHeader: {
    display: "flex",
    flexShrink: 0,
    justifyContent: "space-between",
    margin: 5,
    "& [role='heading']": {
      flexBasis: "100%",
      fontWeight: 700,
      paddingBottom: 0,
      paddingTop: 3,
      paddingLeft: 5,
      marginRight: 8
    },
    "& p[role='heading']": {
      cursor: "pointer"
    },
    "& button": {
      alignSelf: "flex-start",
      // Reusable styles
      borderWidth: 0,
      background: "none",
      outline: "none",
      cursor: "pointer",
      borderRadius: 4,
      padding: 3,
      "&:hover": {
        background: "#D6D6D6"
      }
    },
    "& svg": {
      // Reusable styles
      padding: 3,
      width: "0.8em",
      height: "0.8em",
      color: grey[700]
    }
  }
};

const onOutsideClick = (node, handler) => {
  const onClick = event => {
    if (!node.contains(event.target)) {
      document.body.removeEventListener("click", onClick);
      handler(event);
    }
  };
  document.body.addEventListener("click", onClick);
};

const Title = class extends React.Component {
  constructor(props) {
    super(props);
    this.state = { editing: false };
  }

  componentDidUpdate(prevProps, prevState) {
    if (this.state.editing)
      onOutsideClick(findDOMNode(this), _ => this.setState({ editing: false }));
  }

  render() {
    const props = this.state.editing
      ? {
          component: TextArea,
          value: this.props.text
        }
      : {
          onClick: _ => this.setState({ editing: true }),
          children: this.props.text
        };
    return <Typography role="heading" {...props} />;
  }
};

const View = class extends React.Component {
  constructor(props) {
    super(props);
    this.state = {
      counter: props.cards.length,
      cards: props.cards.map((description, idx) =>
        this.newCard(idx, description)
      )
    };
  }

  newCard(id, description = undefined) {
    return (
      <CardsListCard
        key={id}
        description={description}
        onEditCard={this.props.onEditCard}
      />
    );
  }

  addCard(e) {
    e.preventDefault();
    this.setState((prevState, props) => ({
      counter: prevState.counter + 1,
      cards: [...prevState.cards, this.newCard(prevState.counter + 1)]
    }));
  }

  componentDidUpdate(prevProps, prevState) {
    if (prevState.cards.length < this.state.cards.length)
      this.cardsListEnd.scrollIntoView();
  }

  render() {
    const { classes } = this.props;
    return (
      <Paper
        component="section"
        elevation={1}
        className={classes.container}
        data-testid="CardsList"
      >
        <div className={classes.listHeader}>
          <Title text={this.props.title} />
          <button
            onClick={e => this.props.onEditList(e.currentTarget)}
            aria-describedby={Labels.cardsListActionsMenu.id}
            aria-haspopup={true}
          >
            <MoreHoriz />
          </button>
        </div>
        <div style={{ overflowY: "scroll" }}>
          <ul>{this.state.cards}</ul>
          {/* The sole purpose of this div is to allow scrolling to bottom */}
          <div
            style={{ float: "left", clear: "both" }}
            ref={node => (this.cardsListEnd = node)}
          />
        </div>
        <Button onClick={this.addCard.bind(this)}>Add a card...</Button>
      </Paper>
    );
  }
};

View.propTypes = {
  children: PropTypes.node,
  classes: PropTypes.object.isRequired,
  title: PropTypes.string.isRequired,
  cards: PropTypes.array.isRequired,
  onEditList: PropTypes.func.isRequired,
  onEditCard: PropTypes.func.isRequired
};

View.displayName = path.basename(__filename, path.extname(__filename));

export default withStyles(styles)(View);
