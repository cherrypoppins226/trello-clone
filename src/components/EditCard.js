import React from "react";
import path from "path";
import PropTypes from "prop-types";
import { findDOMNode } from "react-dom";
import { withStyles } from "material-ui/styles";
import Grid from "material-ui/Grid";
import Modal from "material-ui/Modal";
import Typography from "material-ui/Typography";
import Button from "material-ui/Button";
import green from "material-ui/colors/green";
import grey from "material-ui/colors/grey";
import Archive from "@material-ui/icons/Archive";
import Label from "@material-ui/icons/Label";
import Timer from "@material-ui/icons/Timer";
import LibraryBooks from "@material-ui/icons/LibraryBooks";
import ArrowForward from "@material-ui/icons/ArrowForward";
import Person from "@material-ui/icons/Person";
import { cardDescription } from "./CardsListCard";
import TextArea from "./TextArea";

const styles = {
  container: {
    outline: "none",
    pointerEvents: "none"
  },
  editCardDescription: {
    "& textarea": {
      pointerEvents: "all",
      padding: 4,
      paddingLeft: 8,
      // Reusable
      border: 0,
      outline: "none",
      resize: "none",
      borderRadius: 2
    },
    "& button": {
      pointerEvents: "all",
      color: grey[50],
      backgroundColor: green[500],
      textTransform: "none",
      fontWeight: 700,
      marginTop: 15,
      "&:hover": {
        backgroundColor: green[600]
      }
    }
  },
  editCardButtons: {
    "& button": {
      pointerEvents: "all",
      background: "rgba(0, 0, 0, .6)",
      color: grey[50],
      fontWeight: 600,
      textTransform: "none",
      transition: "transform 85ms ease-in",
      "&:hover": {
        background: "rgba(0, 0, 0, .8)",
        transform: "translateX(5px)",
        transition: "transform 85ms ease-in"
      },
      "& svg": {
        padding: 1,
        marginRight: 5,
        width: "0.8em",
        height: "0.8em"
      }
    }
  }
};

export const EDIT_CARD_DESCRIPTION = "edit-card-description";

const View = ({ classes, onClose, card = null, container = null }) => {
  const modal = React.createRef();
  const textarea = React.createRef();

  const modalDidMount = () => {
    const cardCoordinates = card.getBoundingClientRect();
    const modalNode = findDOMNode(modal.current);
    modalNode.style.top = `${cardCoordinates.top}px`;
    modalNode.style.left = `${cardCoordinates.left}px`;
    const textareaNode = findDOMNode(textarea.current);
    textareaNode.style.width = `${cardCoordinates.width}px`;
    textareaNode.style.height = `${cardCoordinates.height + 50}px`;
  };

  return (
    <Modal
      ref={modal}
      aria-describedby={EDIT_CARD_DESCRIPTION}
      open={Boolean(card)}
      container={container || card}
      onClose={onClose}
      onRendered={modalDidMount}
    >
      <Grid container wrap="nowrap" spacing={8} className={classes.container}>
        <div id={EDIT_CARD_DESCRIPTION} style={{ display: "none" }}>
          Edit card: change title, members, archive etc.
        </div>
        <Grid item className={classes.editCardDescription}>
          <Typography
            ref={textarea}
            component={TextArea}
            value={card ? cardDescription(card) : ""}
          />
          <Button variant="raised"> Save </Button>
        </Grid>
        <Grid item className={classes.editCardButtons}>
          <Grid container spacing={8} wrap="nowrap" direction="column">
            {[
              [Label, "Edit Labels"],
              [Person, "Change Members"],
              [ArrowForward, "Move"],
              [LibraryBooks, "Copy"],
              [Timer, "Change Due Date"],
              [Archive, "Archive"]
            ].map(([Icon, text], idx) => (
              <Grid item style={{ paddingBottom: 1 }} key={idx}>
                <Button size="small" variant="flat">
                  <Icon />
                  {text}
                </Button>
              </Grid>
            ))}
          </Grid>
        </Grid>
      </Grid>
    </Modal>
  );
};

View.propTypes = {
  children: PropTypes.node,
  classes: PropTypes.object.isRequired,
  card: PropTypes.object,
  container: PropTypes.object,
  onClose: PropTypes.func.isRequired
};

View.displayName = path.basename(__filename, path.extname(__filename));

export default withStyles(styles)(View);
