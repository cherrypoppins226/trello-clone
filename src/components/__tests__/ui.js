import React from "react";
import snapshotDiff from "snapshot-diff";
import { getByText, getByTestId } from "dom-testing-library";
import { Simulate, fireEvent } from "react-testing-library";
import { renderIntoDocument, NativeEvents } from "../../testHelpers.js";
import { testData } from "../../appData.js";
import App from "../App";
import { cardDescription, EDIT_CARD_LABEL } from "../CardsListCard";
import { LIST_ACTIONS_MENU_LABEL } from "../CardsList";
import { EDIT_CARD_DESCRIPTION } from "../EditCard";

const app = renderIntoDocument(<App lists={testData} />).container
  .firstElementChild;

describe("app", () => {
  it("renders correctly", () => {
    expect(app).toMatchSnapshot();
  });
});

describe("cards list", () => {
  const container = getByTestId(app, "CardsList");

  it("edit title", () => {
    const titleElem = node => node.querySelector("[role='heading']");
    const originalElem = titleElem(container);
    Simulate.click(originalElem);
    expect(titleElem(container).tagName).toBe("TEXTAREA");
    fireEvent(container, NativeEvents.mouse.click);
    expect(titleElem(container)).toEqual(originalElem);
  });

  it("opens list actions menu", () => {
    const actionsMenuButton = container.querySelector(
      `[aria-labelledby="${LIST_ACTIONS_MENU_LABEL}"]`
    );
    const before = document.body.cloneNode(true);
    Simulate.click(actionsMenuButton);
    expect(
      snapshotDiff(before, document.body.cloneNode(true))
    ).toMatchSnapshot();
  });

  it("adds a card", () => {
    const liveList = container.querySelector("ul");
    const countBefore = liveList.childElementCount;
    const lastBefore = liveList.lastElementChild;
    Simulate.click(getByText(container, "Add a card..."));
    expect(liveList.childElementCount).toBe(countBefore + 1);
    expect(liveList.lastElementChild).not.toBe(lastBefore);
    expect(liveList.lastElementChild.tagName).toBe(lastBefore.tagName);
  });
});

describe("cards list card", () => {
  it("edit card modal renders correctly", () => {
    const editCard = app.querySelector(
      `[aria-labelledby="${EDIT_CARD_LABEL}"]`
    );
    Simulate.click(editCard);
    const modal = app.querySelector(
      `[aria-describedby="${EDIT_CARD_DESCRIPTION}"]`
    );
    expect(modal).not.toBeNull();
    const modalContainer = editCard.parentElement;
    expect(modal.parentElement).toBe(modalContainer);
    expect(modal.querySelector("textarea").textContent).toBe(
      cardDescription(modalContainer)
    );
  });
});
