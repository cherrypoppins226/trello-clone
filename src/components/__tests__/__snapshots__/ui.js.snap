// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`app renders correctly 1`] = `
<div
  class="MuiGrid-container MuiGrid-spacing-xs-16 MuiGrid-wrap-xs-nowrap Board-grid-1"
>
  <div
    class="MuiGrid-item Board-cardListContainer-2"
  >
    <section
      class="MuiPaper-root MuiPaper-elevation1 MuiPaper-rounded CardsList-container-93"
      data-testid="CardsList"
    >
      <div
        class="CardsList-listHeader-94"
      >
        <p
          class="MuiTypography-root MuiTypography-body1"
          role="heading"
        >
          Repellat quisquam recusandae alias consequuntur corporis.
        </p>
        <button
          aria-haspopup="true"
          aria-labelledby="open-list-actions-menu"
        >
          <svg
            aria-hidden="true"
            class="MuiSvgIcon-root"
            focusable="false"
            viewBox="0 0 24 24"
          >
            <g>
              <path
                d="M6 10c-1.1 0-2 .9-2 2s.9 2 2 2 2-.9 2-2-.9-2-2-2zm12 0c-1.1 0-2 .9-2 2s.9 2 2 2 2-.9 2-2-.9-2-2-2zm-6 0c-1.1 0-2 .9-2 2s.9 2 2 2 2-.9 2-2-.9-2-2-2z"
              />
            </g>
          </svg>
        </button>
      </div>
      <div
        style="overflow-y: scroll;"
      >
        <ul>
          <li
            class="CardsListCard-container-156"
          >
            <div
              class="MuiPaper-root MuiPaper-elevation1 MuiPaper-rounded CardsListCard-card-157"
            >
              <p
                aria-label="card title"
                class="MuiTypography-root MuiTypography-body1"
              >
                Ratione ut sunt qui amet iure ut libero qui recusandae.
              </p>
              <div
                id="edit-card-label"
                style="display: none;"
              >
                Edit card
              </div>
              <button
                aria-labelledby="edit-card-label"
              >
                <svg
                  aria-hidden="true"
                  class="MuiSvgIcon-root"
                  focusable="false"
                  viewBox="0 0 24 24"
                >
                  <g>
                    <path
                      d="M3 17.25V21h3.75L17.81 9.94l-3.75-3.75L3 17.25zM20.71 7.04c.39-.39.39-1.02 0-1.41l-2.34-2.34c-.39-.39-1.02-.39-1.41 0l-1.83 1.83 3.75 3.75 1.83-1.83z"
                    />
                  </g>
                </svg>
              </button>
            </div>
          </li>
          <li
            class="CardsListCard-container-156"
          >
            <div
              class="MuiPaper-root MuiPaper-elevation1 MuiPaper-rounded CardsListCard-card-157"
            >
              <p
                aria-label="card title"
                class="MuiTypography-root MuiTypography-body1"
              >
                Nulla quam ipsam nobis cupiditate sed dignissimos.
              </p>
              <div
                id="edit-card-label"
                style="display: none;"
              >
                Edit card
              </div>
              <button
                aria-labelledby="edit-card-label"
              >
                <svg
                  aria-hidden="true"
                  class="MuiSvgIcon-root"
                  focusable="false"
                  viewBox="0 0 24 24"
                >
                  <g>
                    <path
                      d="M3 17.25V21h3.75L17.81 9.94l-3.75-3.75L3 17.25zM20.71 7.04c.39-.39.39-1.02 0-1.41l-2.34-2.34c-.39-.39-1.02-.39-1.41 0l-1.83 1.83 3.75 3.75 1.83-1.83z"
                    />
                  </g>
                </svg>
              </button>
            </div>
          </li>
        </ul>
        <div
          style="float: left; clear: both;"
        />
      </div>
      <button
        class="MuiButtonBase-root MuiButton-root"
        tabindex="0"
        type="button"
      >
        <span
          class="MuiButton-label"
        >
          Add a card...
        </span>
        <span
          class="MuiTouchRipple-root"
        />
      </button>
    </section>
  </div>
  <div
    class="MuiGrid-item Board-cardListContainer-2"
  >
    <section
      class="MuiPaper-root MuiPaper-elevation1 MuiPaper-rounded CardsList-container-93"
      data-testid="CardsList"
    >
      <div
        class="CardsList-listHeader-94"
      >
        <p
          class="MuiTypography-root MuiTypography-body1"
          role="heading"
        >
          Incidunt accusantium sed libero repudiandae esse blanditiis natus et eos.
        </p>
        <button
          aria-haspopup="true"
          aria-labelledby="open-list-actions-menu"
        >
          <svg
            aria-hidden="true"
            class="MuiSvgIcon-root"
            focusable="false"
            viewBox="0 0 24 24"
          >
            <g>
              <path
                d="M6 10c-1.1 0-2 .9-2 2s.9 2 2 2 2-.9 2-2-.9-2-2-2zm12 0c-1.1 0-2 .9-2 2s.9 2 2 2 2-.9 2-2-.9-2-2-2zm-6 0c-1.1 0-2 .9-2 2s.9 2 2 2 2-.9 2-2-.9-2-2-2z"
              />
            </g>
          </svg>
        </button>
      </div>
      <div
        style="overflow-y: scroll;"
      >
        <ul>
          <li
            class="CardsListCard-container-156"
          >
            <div
              class="MuiPaper-root MuiPaper-elevation1 MuiPaper-rounded CardsListCard-card-157"
            >
              <p
                aria-label="card title"
                class="MuiTypography-root MuiTypography-body1"
              >
                Velit omnis et porro ut et ipsam explicabo eligendi occaecati.
              </p>
              <div
                id="edit-card-label"
                style="display: none;"
              >
                Edit card
              </div>
              <button
                aria-labelledby="edit-card-label"
              >
                <svg
                  aria-hidden="true"
                  class="MuiSvgIcon-root"
                  focusable="false"
                  viewBox="0 0 24 24"
                >
                  <g>
                    <path
                      d="M3 17.25V21h3.75L17.81 9.94l-3.75-3.75L3 17.25zM20.71 7.04c.39-.39.39-1.02 0-1.41l-2.34-2.34c-.39-.39-1.02-.39-1.41 0l-1.83 1.83 3.75 3.75 1.83-1.83z"
                    />
                  </g>
                </svg>
              </button>
            </div>
          </li>
          <li
            class="CardsListCard-container-156"
          >
            <div
              class="MuiPaper-root MuiPaper-elevation1 MuiPaper-rounded CardsListCard-card-157"
            >
              <p
                aria-label="card title"
                class="MuiTypography-root MuiTypography-body1"
              >
                Et saepe eum dicta eum eaque enim ipsum inventore debitis.
              </p>
              <div
                id="edit-card-label"
                style="display: none;"
              >
                Edit card
              </div>
              <button
                aria-labelledby="edit-card-label"
              >
                <svg
                  aria-hidden="true"
                  class="MuiSvgIcon-root"
                  focusable="false"
                  viewBox="0 0 24 24"
                >
                  <g>
                    <path
                      d="M3 17.25V21h3.75L17.81 9.94l-3.75-3.75L3 17.25zM20.71 7.04c.39-.39.39-1.02 0-1.41l-2.34-2.34c-.39-.39-1.02-.39-1.41 0l-1.83 1.83 3.75 3.75 1.83-1.83z"
                    />
                  </g>
                </svg>
              </button>
            </div>
          </li>
        </ul>
        <div
          style="float: left; clear: both;"
        />
      </div>
      <button
        class="MuiButtonBase-root MuiButton-root"
        tabindex="0"
        type="button"
      >
        <span
          class="MuiButton-label"
        >
          Add a card...
        </span>
        <span
          class="MuiTouchRipple-root"
        />
      </button>
    </section>
  </div>
</div>
`;

exports[`cards list opens list actions menu 1`] = `
Snapshot Diff:
- First value
+ Second value

@@ -1,8 +1,12 @@
- <body>
-   <div>
+ <body
+   style="overflow: hidden; padding-right: 0px;"
+ >
    <div
+     aria-hidden="true"
+   >
+     <div
        class="MuiGrid-container MuiGrid-spacing-xs-16 MuiGrid-wrap-xs-nowrap Board-grid-1"
      >
        <div
          class="MuiGrid-item Board-cardListContainer-2"
        >
@@ -20,10 +24,11 @@
                Repellat quisquam recusandae alias consequuntur corporis.
              </p>
              <button
                aria-haspopup="true"
                aria-labelledby="open-list-actions-menu"
+               aria-owns="cardlist-actions"
              >
                <svg
                  aria-hidden="true"
                  class="MuiSvgIcon-root"
                  focusable="false"
@@ -265,6 +270,177 @@
            </button>
          </section>
        </div>
      </div>
    </div>
+   <div
+     class="MuiModal-root CardsListActionsMenu-container-152"
+   >
+     <div
+       aria-hidden="true"
+       class="MuiBackdrop-root MuiBackdrop-invisible"
+       style="opacity: 1; -webkit-transition: opacity 225ms cubic-bezier(0.4, 0, 0.2, 1) 0ms;"
+     />
+     <div
+       class="MuiPaper-root MuiPaper-elevation1 MuiPaper-rounded MuiPopover-paper"
+       role="document"
+       style="opacity: 1; top: 16px; left: 16px;"
+       tabindex="-1"
+     >
+       <p
+         class="MuiTypography-root MuiTypography-body1 MuiTypography-alignCenter"
+         role="heading"
+       >
+         List Actions
+       </p>
+       <ul
+         class="MuiList-root MuiList-dense MuiList-padding"
+         role="menu"
+       >
+         <hr
+           class="MuiDivider-root"
+           tabindex="0"
+         />
+         <li
+           class="MuiButtonBase-root MuiListItem-root MuiListItem-dense MuiListItem-gutters MuiListItem-button MuiMenuItem-root"
+           role="menuitem"
+           tabindex="-1"
+         >
+           <p
+             class="MuiTypography-root MuiTypography-body1"
+           >
+              
+             Add Card...
+              
+           </p>
+           <span
+             class="MuiTouchRipple-root"
+           />
+         </li>
+         <li
+           class="MuiButtonBase-root MuiListItem-root MuiListItem-dense MuiListItem-gutters MuiListItem-button MuiMenuItem-root"
+           role="menuitem"
+           tabindex="-1"
+         >
+           <p
+             class="MuiTypography-root MuiTypography-body1"
+           >
+              
+             Copy List...
+              
+           </p>
+           <span
+             class="MuiTouchRipple-root"
+           />
+         </li>
+         <li
+           class="MuiButtonBase-root MuiListItem-root MuiListItem-dense MuiListItem-gutters MuiListItem-button MuiMenuItem-root"
+           role="menuitem"
+           tabindex="-1"
+         >
+           <p
+             class="MuiTypography-root MuiTypography-body1"
+           >
+              
+             Move List...
+              
+           </p>
+           <span
+             class="MuiTouchRipple-root"
+           />
+         </li>
+         <li
+           class="MuiButtonBase-root MuiListItem-root MuiListItem-dense MuiListItem-gutters MuiListItem-button MuiMenuItem-root"
+           role="menuitem"
+           tabindex="-1"
+         >
+           <p
+             class="MuiTypography-root MuiTypography-body1"
+           >
+              
+             Watch
+              
+           </p>
+           <span
+             class="MuiTouchRipple-root"
+           />
+         </li>
+         <hr
+           class="MuiDivider-root"
+           tabindex="-1"
+         />
+         <li
+           class="MuiButtonBase-root MuiListItem-root MuiListItem-dense MuiListItem-gutters MuiListItem-button MuiMenuItem-root"
+           role="menuitem"
+           tabindex="-1"
+         >
+           <p
+             class="MuiTypography-root MuiTypography-body1"
+           >
+              
+             Sort By
+              
+           </p>
+           <span
+             class="MuiTouchRipple-root"
+           />
+         </li>
+         <hr
+           class="MuiDivider-root"
+           tabindex="-1"
+         />
+         <li
+           class="MuiButtonBase-root MuiListItem-root MuiListItem-dense MuiListItem-gutters MuiListItem-button MuiMenuItem-root"
+           role="menuitem"
+           tabindex="-1"
+         >
+           <p
+             class="MuiTypography-root MuiTypography-body1"
+           >
+              
+             Move All Cards in This List...
+              
+           </p>
+           <span
+             class="MuiTouchRipple-root"
+           />
+         </li>
+         <li
+           class="MuiButtonBase-root MuiListItem-root MuiListItem-dense MuiListItem-gutters MuiListItem-button MuiMenuItem-root"
+           role="menuitem"
+           tabindex="-1"
+         >
+           <p
+             class="MuiTypography-root MuiTypography-body1"
+           >
+              
+             Archive All Cards in This List...
+              
+           </p>
+           <span
+             class="MuiTouchRipple-root"
+           />
+         </li>
+         <hr
+           class="MuiDivider-root"
+           tabindex="-1"
+         />
+         <li
+           class="MuiButtonBase-root MuiListItem-root MuiListItem-dense MuiListItem-gutters MuiListItem-button MuiMenuItem-root"
+           role="menuitem"
+           tabindex="-1"
+         >
+           <p
+             class="MuiTypography-root MuiTypography-body1"
+           >
+              
+             Archive This List
+              
+           </p>
+           <span
+             class="MuiTouchRipple-root"
+           />
+         </li>
+       </ul>
+     </div>
+   </div>
  </body>
`;
